Class Crawler.py

```diff
@@ -0,0 +1,168 @@
+import requests
+import re
+from bs4 import BeautifulSoup
+import random
+import datetime
+random.seed(datetime.datetime.now())
+
+    
+class Crawler(object):
+    def __init__(self, address):
+        
+        self.address = address
+        self.html = requests.get(address)
+        self.soup = BeautifulSoup(self.html.content, "lxml")
+        
+    
+    def getAddress(self):
+        
+        return self.address
+        
+    def domainName(self):
+        '''
+        return domain name of given website
+        '''
+        address = self.address
+        if "https://" in address:
+            address = address.replace("https://", "")
+        if "http://" in address:
+            address = address.replace("http://", "")
+        if "www." in address:
+            address = address.replace("www.", "")
+        domain = address.split('.')[0]
+        return domain
+
+    def getLinks(self):
+        '''
+        returns list of links (strings) that appears on given website
+        '''
+        links_on_site = []
+        soup = self.soup
+        for link in  soup.findAll('a', attrs={'href': re.compile("^(http|www)")}):
```

Comment:
You may use `self.soup` directly in this loop. No binding to `soup` variable is needed since you are not changing anything in `self.soup`